ops/terraform/services/common/scripts/destroy-autoscaled-nodes.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# Terraform helper script called as a local-exec provisioner at destroy-time of the Aurora cluster
+# resource. This script will enumerate the autoscaled reader nodes within the given Aurora cluster
+# (provided by environment variable $DB_CLUSTER_ID), mark them all for deletion, and then wait
+# indefinitely for all nodes to be successfully deleted. This script is intended to run ONLY in
+# ephemeral environments and will immediately fail if it is run in an established environment.
+# Environment variables:
+#   DB_CLUSTER_ID: The cluster identifier of the ephemeral database cluster that is being destroyed
+# . BFD_ENVIRONMENT: The name of the environment that this script is being run against
+
+set -euo pipefail
+
+# Ensure that the BFD_ENVIRONMENT environment var is specified and not empty
+trimmed_bfd_env="$(echo "$BFD_ENVIRONMENT" | tr -d '[:space:]')"
+if [[ -z $trimmed_bfd_env ]]; then
+  echo "BFD_ENVIRONMENT must not be an empty string or whitespace"
+  exit 1
+fi
+
+# Ensure that whatever environment this script is running in is NOT an established environment. We
+# never want to destroy nodes in our established environments
+if [[ $trimmed_bfd_env == "prod" || $trimmed_bfd_env == "prod-sbx" || $trimmed_bfd_env == "test" ]]; then
+  echo "Cannot destroy nodes in established environment $BFD_ENVIRONMENT"
+  exit 1
+fi
+
+# Be _extra_ careful about the cluster ID given by the $DB_CLUSTER_ID environment variable.
+# Specifically, remove all whitespace (which is invalid for a cluster ID) and ensure that the
+# resulting, trimmed ID is non-empty before even making any deletion attempts as the AWS CLI will
+# treat an empty cluster ID as equivalent to "give me all clusters" when calling
+# describe-db-clusters
+trimmed_cluster_id="$(echo "$DB_CLUSTER_ID" | tr -d '[:space:]')"
+if [[ -z $trimmed_cluster_id ]]; then
+  echo "DB_CLUSTER_ID must not be an empty string or whitespace"
+  exit 1
+fi
+
+# jq expression returns the first element in the returned array as a precaution against a scenario
+# where multiple clusters are erroneously returned. Otherwise, the nodes in all clusters would be
+# marked for deletion. Additionally, we select for only the nodes with names containing
+# "autoscaling" to avoid possibly deleting the writer node which is managed by Terraform directly
+nodes_in_cluster="$(
+  aws rds describe-db-clusters --db-cluster-identifier "$trimmed_cluster_id" |
```

Comment:
I realize we have a number of defaults that likely cover this item, but would it be prudent to include "--output json" on all of the CLI commands as a safety precaution?

Reply:
This is definitely reasonable; being explicit is always a good idea. I'll make this change.

Related: I do think there are quite a few places where the AWS CLI is invoked and `--output json` is assumed without being explicitly specified, so maybe this is part of a wider discussion of whether we should trust that `--output json` is always the default wherever we run the AWS CLI? I think it's reasonable for us to continue to assume that, but maybe going forward we should always specify the output format explicitly.

Reply:
Done in [e5f63dc](https://github.com/CMSgov/beneficiary-fhir-data/pull/2393/commits/e5f63dc9945afd6cd523ada185885a45a3e009b8).